samtranslator/model/eventbridge_utils.py

```diff
@@ -41,12 +41,18 @@ def validate_dlq_config(source_logical_id, dead_letter_config):
             raise InvalidEventException(source_logical_id, "No 'Arn' or 'Type' property provided for DeadLetterConfig")
 
     @staticmethod
-    def get_dlq_queue_arn_and_resources(cw_event_source, source_arn, attributes):
+    def get_dlq_queue_arn_and_resources(cw_event_source, source_arn, attributes, intrinsics_resolver):
         """returns dlq queue arn and dlq_resources, assuming cw_event_source.DeadLetterConfig has been validated"""
         dlq_queue_arn = cw_event_source.DeadLetterConfig.get("Arn")
         if dlq_queue_arn is not None:
             return dlq_queue_arn, []
         queue_logical_id = cw_event_source.DeadLetterConfig.get("QueueLogicalId")
+        if (
+            isinstance(queue_logical_id, dict)
+            and list(queue_logical_id.keys())[0] in intrinsics_resolver.supported_intrinsics
+        ):
+            queue_logical_id = intrinsics_resolver.resolve_parameter_refs(queue_logical_id)
+            queue_logical_id = list(queue_logical_id.values())[0]
```

Comment:
Why resolve within this function instead of somewhere more central? The more we adhoc resolving in a bunch of places, the harder this will be to maintain over the long run.